io/shared/src/test/scala/fs2/io/file/FilesSuite.scala

```diff
@@ -555,5 +562,149 @@ class FilesSuite extends Fs2Suite with BaseFileSuite {
         .use(Files[IO].isRegularFile(_))
         .assertEquals(false)
     }
+
+    test("returns false if the path does not exist") {
+      tempDirectory
+        .map(_.resolve("non-existent-file"))
+        .use(Files[IO].isRegularFile(_))
+        .assertEquals(false)
+    }
+  }
+
+  group("isReadable") {
+    test("returns true if the path is for a readable file") {
+      tempFile
+        .use(Files[IO].isReadable(_))
+        .assertEquals(true)
+    }
+
+    test("returns true if the path is for a directory") {
+      tempDirectory
+        .use(Files[IO].isReadable(_))
+        .assertEquals(true)
+    }
+
+    test("returns false if the path does not exist") {
+      tempDirectory
+        .map(_.resolve("non-existent-file"))
+        .use(Files[IO].isReadable(_))
+        .assertEquals(false)
+    }
+  }
+
+  group("isWritable") {
+    test("returns true if the path is for a readable file") {
+      tempFile
+        .use(Files[IO].isWritable(_))
+        .assertEquals(true)
+    }
+
+    test("returns true if the path is for a directory") {
+      tempDirectory
+        .use(Files[IO].isWritable(_))
+        .assertEquals(true)
+    }
+
+    test("returns false if the path does not exist") {
+      tempDirectory
+        .map(_.resolve("non-existent-file"))
+        .use(Files[IO].isWritable(_))
+        .assertEquals(false)
+    }
+  }
+
+  group("isHidden") {
```

Comment:
Even though there's no windows CI, this will be annoying for anyone developing on Windows. Not sure how big a deal that is.

Reply:
Yeah 🤔 
If someone's working on fs2 in Windows and runs the tests.. that will be annoying (I kind of blanked out such a possibility :/ )
I wonder if we can do some platform checks in the test code and, at least, always pass `isHidden` true-case if it's windows?

Reply:
> Eh, it's merged. Let's see if anyone complains :P
 
=)

> The problem is we need cross-platform (JVM/JS) checks for checking the platform (Unix/Windows). Possible, but annoying!

Should be non-crazy enough to implement, right?

Is there a good place for it? (like `Files[F]` is for the file stuff)

Reply:
Oh, as a public API? Probably not in fs2. I was thinking as an internal/test utility.